x/babylon/keeper/handler_plugin.go

```diff
@@ -64,11 +67,40 @@ func (h CustomMsgHandler) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddre
 		return nil, nil, nil, sdkerrors.ErrUnauthorized.Wrapf("contract has no permission for Babylon operations")
 	}
 
-	return h.handleTestMsg(ctx, contractAddr, customMsg.Test)
+	return h.handleMintRewardsMsg(ctx, contractAddr, customMsg.MintRewards)
 }
 
-func (h CustomMsgHandler) handleTestMsg(ctx sdk.Context, actor sdk.AccAddress, testMsg *contract.TestMsg) ([]sdk.Event, [][]byte, [][]*codectypes.Any, error) {
-	return []sdk.Event{}, nil, nil, nil
+func (h CustomMsgHandler) handleMintRewardsMsg(ctx sdk.Context, actor sdk.AccAddress, mintMsg *contract.MintRewardsMsg) ([]sdk.Event, [][]byte, [][]*codectypes.Any, error) {
+	coin, err := wasmkeeper.ConvertWasmCoinToSdkCoin(mintMsg.Amount)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	params := h.k.GetParams(ctx)
+	// Validate actor
+	if actor.String() != params.BtcFinalityContractAddress {
+		return nil, nil, nil, sdkerrors.ErrUnauthorized.Wrapf("minter must be the finality contract")
+	}
+
+	// Define recipient
+	recipient, err := sdk.AccAddressFromBech32(mintMsg.Recipient)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	rewards, err := h.k.MintBlockRewards(ctx, recipient, coin)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return []sdk.Event{sdk.NewEvent(
+		types.EventTypeMintRewards,
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		sdk.NewAttribute(sdk.AttributeKeySender, actor.String()),
+		sdk.NewAttribute(sdk.AttributeKeyAmount, rewards.String()),
+	)}, nil, nil, nil
```

Comment:
Events are not needed as Cosmos SDK has already done it https://github.com/cosmos/cosmos-sdk/blob/8bfcf554275c1efbb42666cc8510d2da139b67fa/x/bank/keeper/send.go#L238-L249